packages/hub/services/scheduled-payments/fetcher.ts

```diff
@@ -99,7 +99,12 @@ export default class ScheduledPaymentsFetcherService {
         ORDER BY last_failed_payment_attempt.started_at ASC NULLS FIRST
         LIMIT ${limit};`;
 
-    return prisma.scheduledPayment.findMany({ where: { id: { in: results.map((result) => result.id) } } });
+    let ids = results.map((result) => result.id);
+
+    // Make sure the collection is sorted in the same order as the ids array
+    return (await prisma.scheduledPayment.findMany({ where: { id: { in: ids } } })).sort(
```

Comment:
Don't we want to sort scheduled payments by the last failed payment attempt, not id?

Reply:
We do, the `results` are sorted by that.

Then we just make sure the prisma models have the same order as results